core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -753,10 +786,17 @@ object Doc {
    *     doc.render(10) // produces "1, 2, 3"
    */
   def fill(sep: Doc, ds: Iterable[Doc]): Doc = {
+    import FlattenResult._
 
     val flatSep = sep.flatten
     val sepGroup = sep.grouped
 
+    def unionIfFlattenable(first: Doc, second: Doc): Doc = first.flattenResult match {
```

Comment:
wait, if first is unflattenable we are just discarding it completely? This does not seem intuitive to me.

I think we will need considerably more tests and laws for `fill`.

Reply:
Agreed.  Let's pause this until we get the tests for `fill` more precise.  At least I want to come up with some tests that make `fill` obviously different from `intercalate` so it's clearly useful.  The Haskell libraries don't have this version of `fill` at all.

Reply:
Yeah, this was me enforcing the overly aggressive `Union` invariant.  I'm relaxing this.